### PR TITLE
[RW-5379][risk=no] move mappers from CohortReviewController to CohortReviewService

### PR DESCRIPTION
**Pull request by freemabd**
Cleanup of CohortReviewController
1. Moved mappers into CohortReviewServiceImpl
2. Separated Controller level exceptions from Service level exceptions
3. Fixed a sorting UI issue in review using FilterColumns.SEXATBIRTH

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/CohortAnnotationDefinitionController.java [hunk 1]
```diff
@@ -134,11 +132,7 @@ private CohortAnnotationDefinition findCohortAnnotationDefinition(
   }
 
   private void validateCohortExist(long cohortId) {
-    DbCohort cohort = cohortReviewService.findCohort(cohortId);
-    if (cohort == null) {
-      throw new NotFoundException(
-          String.format("Not Found: No Cohort exists for cohortId: %s", cohortId));
-    }
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/CohortAnnotationDefinitionController.java:141]: Remove this cause it's also happening in cohortReviewService.findCohort(cohortId)

File: api/src/main/java/org/pmiops/workbench/cohortreview/mapper/CohortReviewMapper.java [hunk 2]
```diff
@@ -28,25 +28,19 @@
...
-  // used for pagination. Effectively deprecated, to remove with RW-4706
-  @Mapping(target = "queryResultSize", ignore = true)
-  @Mapping(target = "page", ignore = true)
-  @Mapping(target = "pageSize", ignore = true)
-  @Mapping(target = "sortOrder", ignore = true)
-  @Mapping(target = "sortColumn", ignore = true)
-  // this fetches all participants, and can be large, we don't want to fetch by
-  // default. May be removed from object pending design
-  @Mapping(target = "participantCohortStatuses", ignore = true)
-  CohortReview dbModelToClient(DbCohortReview dbCohortReview, @Context PageRequest pageRequest);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cohortreview/mapper/CohortReviewMapper.java:41]: No longer need this mapper method. I will create a future story that will remove page, pageSize, sortOrder and sortColumn for the CohortReview object

File: api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java [hunk 3]
```diff
@@ -219,19 +199,16 @@
...
-                    participantCohortStatusMapper.dbModelToClient(
-                        pcs, cohortBuilderService.findAllDemographicsMap()))
-            .collect(Collectors.toList()));
-    return ResponseEntity.ok(responseReview);
+    cohortReview
+        .page(pageRequest.getPage())
+        .pageSize(pageRequest.getPageSize())
+        .sortOrder(pageRequest.getSortOrder().toString())
+        .sortColumn(pageRequest.getSortColumn())
+        .participantCohortStatuses(paginatedPCS);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java:210]: I will create a future story that will remove page, pageSize, sortOrder and sortColumn for the CohortReview object

File: api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java [hunk 4]
```diff
@@ -99,70 +92,56 @@
...
 
   /**
    * Create a cohort review per the specified workspaceId, cohortId, cdrVersionId and size. If
    * participant cohort status data exists for a review or no cohort review exists for
    * cohortReviewId then throw a {@link BadRequestException}.
-   *
-   * @param workspaceNamespace
-   * @param workspaceId
-   * @param cohortId
-   * @param cdrVersionId
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java:161]: Why is this removed?

File: api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java [hunk 5]
```diff
@@ -110,86 +105,118 @@ public DbCohortReview findCohortReview(Long cohortId, Long cdrVersionId) {
...
-      DbCohortReview cohortReview, List<DbParticipantCohortStatus> participantCohortStatuses) {
-    saveCohortReview(cohortReview);
+      CohortReview cohortReview, List<DbParticipantCohortStatus> participantCohortStatuses) {
+    cohortReviewDao.save(cohortReviewMapper.clientToDbModel(cohortReview));
     participantCohortStatusDao.saveParticipantCohortStatusesCustom(participantCohortStatuses);
   }
 
+  public CohortReview updateCohortReview(
+      CohortReview cohortReview, Long cohortReviewId, Timestamp lastModified) {
+    DbCohortReview dbCohortReview = findDbCohortReview(cohortReviewId);
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java:150]: Should we wrap this around a try catch just in case cohortReviewId does not exist in DB

File: api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java [hunk 6]
```diff
@@ -110,86 +105,118 @@ public DbCohortReview findCohortReview(Long cohortId, Long cdrVersionId) {
...
+    }
+  }
+
   @Override
-  public DbParticipantCohortStatus saveParticipantCohortStatus(
-      DbParticipantCohortStatus participantCohortStatus) {
-    return participantCohortStatusDao.save(participantCohortStatus);
+  public ParticipantCohortStatus updateParticipantCohortStatus(
+      Long cohortReviewId, Long participantId, CohortStatus status, Timestamp lastModified) {
+    DbCohortReview dbCohortReview = findDbCohortReview(cohortReviewId);
```
Comment (NehaBroad) [inline api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java:175]: Similarly here

File: api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java (hunk 5 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java:150]: I take it back i see findDbCohortReview is already throwing that execption

Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java:150]: Nvm i see findDbCohortReview is already doing that

File: api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java (hunk 6 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java:175]: nvm findDbCohortReview is taking care of it

File: api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java (hunk 4 above)
Comment (freemabd) [reply api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java:161]: A few reasons, they didn't match the parameters being passed into the method and Intellij gives a warning for any `@param` that doesn't have a description attached to them.  I feel that they are self explanatory and don't need descriptions and was tired of seeing the warnings, so i removed them.
